crates/task-impls/src/consensus.rs

```diff
@@ -295,7 +295,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
     }
 
     #[instrument(skip_all, fields(id = self.id, view = *self.cur_view), name = "Consensus vote if able", level = "error")]
-
+    // Check if we are able to vote, like whether the proposal is valid, whether we have DAC and VID share, and if so, vote
```

Comment:
Agree it's too long. Let me see whether there's any easy cleanup.
- Line 356 is an action for genesis view, it will not check VID share nor DA cert.
- Those two same lines, one is for creating vote for genesis view, which will `return` after the vote, therefore we need the other one for the normal cases.

Reply:
Not that easy to turn those common parts into a function :( Maybe for a future issue. I added more comments for the clarification of this function.